api-java/src/main/java/javaclasses/mealorder/c/po/PurchaseOrders.java

```diff
@@ -49,57 +56,66 @@ private PurchaseOrders() {
+     * <p> Takes order list and purchase order identifier from {@code cmd}.
+     * Checks each order in the order list to match the purchase order date
+     * and vendor. Also checks for orders with an empty dish list and orders with
      * not {@code 'ORDER_ACTIVE'} status.
      *
-     * @param cmd command to create a purchase order.
-     * @return is allowed a purchase order creation.
+     * @param cmd command to create a purchase order
+     * @return {@code true} if a purchase order creation is allowed
      */
-    public static boolean isAllowedPurchaseOrderCreation(CreatePurchaseOrder cmd) {
+    static boolean isAllowedPurchaseOrderCreation(CreatePurchaseOrder cmd) {
         checkNotNull(cmd);
         final PurchaseOrderId purchaseOrderId = cmd.getId();
-        List<Order> ordersList = cmd.getOrderList();
-
-        final boolean result = ordersList.stream()
-                                         .allMatch(o -> doesOrderBelongToPO(purchaseOrderId, o));
-        return !ordersList.isEmpty() && result;
-    }
+        final List<Order> ordersList = cmd.getOrderList();
 
-    private static boolean doesOrderBelongToPO(PurchaseOrderId purchaseOrderId, Order order) {
-        final VendorId purchaseOrderVendorId = purchaseOrderId.getVendorId();
-        final LocalDate purchaseOrderDate = purchaseOrderId.getPoDate();
-
-        if (!(checkOrderIsActive(order) && checkOrderingDatesMatch(order, purchaseOrderDate))) {
-            return false;
-        }
-        return checkOrderNotEmpty(order) && checkVendorsMatch(order, purchaseOrderVendorId);
+        final boolean orderListNotEmpty = !ordersList.isEmpty();
+        final boolean ordersFitToPO = ordersList.stream()
+                                                .allMatch(
+                                                        o -> doesOrderFitToPO(purchaseOrderId, o));
+        return orderListNotEmpty && ordersFitToPO;
     }
 
     /**
      * Finds orders which contain more than {@code MAX_SINGLE_DISH_COUNT}
      * equal dishes.
      *
-     * <p>Those orders are considered invalid.
+     * <p> Those orders are considered invalid and returned as result.
      *
-     * @param orders list to check.
-     * @return list of invalid orders.(Empty if all orders are valid)
+     * @param orders the list to check
+     * @return list of invalid orders, empty if all orders are valid
      */
-    public static List<Order> findInvalidOrders(List<Order> orders) {
+    static List<Order> findInvalidOrders(List<Order> orders) {
         checkNotNull(orders);
         final List<Order> invalidOrders = orders.stream()
                                                 .filter(o -> !isOrderValid(o))
                                                 .collect(Collectors.toList());
         return invalidOrders;
     }
 
-    public static boolean hasInvalidOrders(List<Order> orders) {
+    /**
+     * Searches for orders which contain more than {@code MAX_SINGLE_DISH_COUNT}
```

Comment:
Checks if purchase order contains any orders that contain more than {@code MAX_SINGLE_DISH_COUNT}